File: VB/WindowsApplication3/MainForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors


Namespace DXSample
	Partial Public Class MainForm
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Private extender As TreeListColumnMenuExtender

		Protected Overrides Sub OnLoad(ByVal e As EventArgs)
			MyBase.OnLoad(e)
			Dim TempXViews As DevExpress.XtraTreeList.Design.XViews = New DevExpress.XtraTreeList.Design.XViews(treeList1)
			extender = New TreeListColumnMenuExtender(treeList1)
			extender.EnableRenameColumnOption()
		End Sub

		Protected Overrides Sub OnFormClosing(ByVal e As FormClosingEventArgs)
			MyBase.OnFormClosing(e)
			extender.DisableRenameColumnOption()
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication3/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;


namespace DXSample {
    public partial class MainForm: XtraForm {
        public MainForm() {
            InitializeComponent();
        }

        TreeListColumnMenuExtender extender;
      
        protected override void OnLoad(EventArgs e) {
            base.OnLoad(e);
            new DevExpress.XtraTreeList.Design.XViews(treeList1);
            extender = new TreeListColumnMenuExtender(treeList1);
            extender.EnableRenameColumnOption();
        }

        protected override void OnFormClosing(FormClosingEventArgs e) {
            base.OnFormClosing(e);
            extender.DisableRenameColumnOption();
        }
    }
}
